hibernate-core/src/main/java/org/hibernate/internal/util/collections/ArrayHelper.java

```diff
@@ -428,13 +407,8 @@ public static int countNonNull(Serializable[] array) {
 	}
 
 	public static String[] reverse(String[] source) {
-		final int length = source.length;
-		final String[] destination = new String[length];
-		for ( int i = 0; i < length; i++ ) {
-			final int x = length - i - 1;
-			destination[x] = source[i];
-		}
-		return destination;
+		Collections.reverse( Arrays.asList( source ) );
```

Comment:
I checked it and this method is not even used, so we might want to remove it anyway.